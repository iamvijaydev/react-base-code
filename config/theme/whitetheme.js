import colors from './colors';
import fonts from './fonts';
import zindex from './zindex';
import media from './media';

const whitetheme = {

  /* make all colors avaliable */
  colors,

  /* human readable, meaningful, z-index values */
  zindex,

  /* media rules */
  media,

  /* dark theme */
  backgroudColor: colors.white,
  textColor: colors.gray900,

  /* for UI elements */
  primaryColor: colors.green500,
  secondaryColor: colors.yellow500,
  alternativeColor: colors.red500,

  /* status colors */
  errorColor: colors.red500,
  successColor: colors.green500,
  notifColor: colors.green500,
  neutralColor: colors.gray200,

  /* link colors */
  linkColor: colors.blue500,
  linkActiveColor: colors.blue600,
  linkVisitedColor: colors.indigo500,

  /* theme specific fonts */
  primaryFont: fonts.sans,
  secondaryFont: fonts.serif,
  alternativeFont: fonts.mono,

  /* font weights */
  bold: fonts.bold,
  medium: fonts.medium,
  light: fonts.light,
};

export { whitetheme as default };
