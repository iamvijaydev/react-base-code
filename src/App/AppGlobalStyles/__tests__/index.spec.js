import React from 'react';
import { shallow } from 'enzyme';

import GlobalStyles from '../index';
import whitetheme from '../../../../config/theme/whitetheme';

describe('<GlobalStyles /> component', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallow(<GlobalStyles theme={whitetheme} />);
  });

  afterAll(() => {
    if (wrapper) wrapper.unmount();
  });

  it('should be defined', () => {
    expect(GlobalStyles).toBeDefined();
  });

  it('snapshot ok', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render nothing', () => {
    expect(wrapper.get(0)).toBe(null);
  });

  it('should inject styles', () => {
    expect(document.getElementById('GLOBAL_STYLES').innerHTML).not.toHaveLength(0);
  });
});
