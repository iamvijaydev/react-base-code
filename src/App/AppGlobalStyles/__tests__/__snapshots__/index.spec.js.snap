// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalStyles /> component snapshot ok 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GlobalStyles
    theme={
      Object {
        "alternativeColor": "#f05323",
        "alternativeFont": "'Roboto Mono', monospace",
        "backgroudColor": "#fff",
        "bold": 700,
        "colors": Object {
          "black": "#000",
          "gray100": "#eee",
          "gray200": "#aaa",
          "gray300": "#555",
          "gray400": "#111",
          "green": "#36ba4e",
          "red": "#f05323",
          "white": "#fff",
          "yellow": "#ffbf00",
        },
        "errorColor": "#f05323",
        "light": 100,
        "linkActiveColor": "#ffbf00",
        "linkColor": "#f05323",
        "linkVisitedColor": "#36ba4e",
        "media": Object {
          "desktop": "@media only screen and (min-width: 992px)",
          "tablet": "@media only screen and (min-width: 768px)",
          "tabletOnly": "@media only screen and (min-width: 768px) and (max-width: 991px)",
        },
        "medium": 400,
        "neutralColor": "#aaa",
        "notifColor": "#36ba4e",
        "primaryColor": "#36ba4e",
        "primaryFont": "'Open Sans', sans-serif",
        "secondaryColor": "#ffbf00",
        "secondaryFont": "'Slabo 27px', serif",
        "successColor": "#36ba4e",
        "textColor": "#111",
        "zindex": Object {
          "dropdown": 96,
          "modal": 98,
          "modalx": 99,
          "notif": 95,
          "tooltip": 97,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
